### PR TITLE
Fixed [BUG] Directions Waypoint overflow #99

### LINKED ISSUE #99: [BUG] Directions Waypoint overflow
### Description

List of waypoints extend outside the box.

I think we should make the list of waypoints scrollable instead. I would like to work on it for GSoC.

### Screenshots

overflow

### PR DESCRIPTION
**Pull request by Ghat0tkach**
## 🛠️ Fixes Issue
Closes #99 

## 👨‍💻 Changes proposed
Since the directions were overflowing whenever we get more than 6 waypoints(checked across all devices) , so whenever there are more than 6 waypoints , the overflow automatically gets set to auto. Hence the problem is solved. 
It Also increases the height of the box to look User-Friendly.

## 📷 Screenshots

https://user-images.githubusercontent.com/59855919/223726603-70ccae72-d305-48f5-8348-6327c38519a1.mp4

### CODE REVIEW

File: src/Controls/Directions/Waypoints/index.jsx [hunk 1]
```diff
@@ -34,7 +34,14 @@ const getListStyle = (isDraggingOver) => ({
   paddingBottom: 20,
   maxHeight: 350,
   height: 300,
-  oveflow: 'inherit',
+})
+const ExtendedListStyle = (isDraggingOver) => ({
```
Comment (nilsnolde) [inline src/Controls/Directions/Waypoints/index.jsx:38]: let's keep the code "style" (if there is any..) and make it camelCase

File: src/Controls/Directions/Waypoints/index.jsx [hunk 2]
```diff
@@ -83,16 +90,24 @@ class Waypoints extends Component {
 
   render() {
     const { waypoints } = this.props.directions
+    const isExtendedList = waypoints.length > 6 || waypoints.length === 6
```
Comment (nilsnolde) [inline src/Controls/Directions/Waypoints/index.jsx:93]: JS surely has a concept of >=?

File: src/Controls/Directions/Waypoints/index.jsx (hunk 1 above)
Comment (nilsnolde) [reply src/Controls/Directions/Waypoints/index.jsx:38]: in fact you could merge this with the function above right? I mean it's basically a lambda. Just accept `isExtendedList`  as an argument and make the values of `maxheight`, `height` a ternary depending on `isExtendedList`. less code and a tiny bit more readable

**Review by nilsnolde — CHANGES_REQUESTED** [review]
Thanks, looks great! Some nits

File: src/Controls/Directions/Waypoints/index.jsx (hunk 2 above)
Comment (Ghat0tkach) [reply src/Controls/Directions/Waypoints/index.jsx:93]: Oh yes , actually i was doing some experiments with the code .Actually  I wished to set the overflow auto only whenever the waypoints take whole up the available screen size. However this didn't go as planned . So i reverted back to the original😅

File: src/Controls/Directions/Waypoints/index.jsx (hunk 1 above)
Comment (Ghat0tkach) [reply src/Controls/Directions/Waypoints/index.jsx:38]: Oh yes , Sorry for the inconvenience.
Will soon fix it.
Thank you

File: src/Controls/Directions/Waypoints/index.jsx [hunk 3]
```diff
@@ -83,16 +81,20 @@ class Waypoints extends Component {
...
               <div
-                className={`flex flex-column`}
+                className={`flex 
+                ${
+                  isExtendedList ? 'flex-column overflow-auto' : 'flex-column'
+                }`}
                 {...provided.droppableProps}
                 ref={provided.innerRef}
-                style={getListStyle(snapshot.isDraggingOver)}
+                 style={getListStyle(snapshot.isDraggingOver)}
```
Comment (nilsnolde) [inline src/Controls/Directions/Waypoints/index.jsx:97]: you forgot to actually pass the argument;)

Comment (Ghat0tkach) [reply src/Controls/Directions/Waypoints/index.jsx:97]: but i do test the application and it worked

Comment (Ghat0tkach) [reply src/Controls/Directions/Waypoints/index.jsx:97]: iam so sorry , often in excitement . I do miss these things T-T

**Review by nilsnolde — APPROVED** [review]

**Pull request merged**